src/utils/viewer/viewer.cpp

```diff
@@ -612,6 +616,22 @@ void addSensors(esp::agent::AgentConfiguration& agentConfig, bool isOrtho) {
   // add the equirectangular semantic sensor
   addEquirectangularSensor("semantic_equirectangular",
                            esp::sensor::SensorType::Semantic);
+
+// #ifdef ESP_BUILD_WITH_AUDIO // todo sangarg : figure out why this flag is not getting set
+  // add audio sensor
+  auto addAudioSensor = [&](const std::string& uuid,
```

Comment:
Question : Once the audio sensor is added, figure out how the getObservation etc functions are triggered
Figure out source location and how to add new query for triggering the audio sensor for the agent
Also check what is the expected behavior for the agent. Do we just send the floats buffer? Is the agent supposed to move or does all that happen outside this repo.